src/irc/core/irc-cap.c

```diff
@@ -79,73 +79,148 @@ static void cap_emit_signal (IRC_SERVER_REC *server, char *cmd, char *args)
 	g_free(signal_name);
 }
 
+static gboolean parse_cap_name(char *name, char **key, char **val)
+{
+	g_return_val_if_fail(name != NULL, FALSE);
+	g_return_val_if_fail(name[0] != '\0', FALSE);
+
+	const char *eq = strchr(name, '=');
+	/* KEY only value */
+	if (eq == NULL) {
+		*key = g_strdup(name);
+		*val = NULL;
+		return TRUE;
+	}
```

Comment:
I also like this code style but until we get the fully automated code formatter working please adjust to `} else if {` on a single line